go-sdk/pkg/sdk/keptn.go

```diff
@@ -314,3 +329,19 @@ func (k *Keptn) createErrorFinishedEventForTriggeredEvent(event cloudevents.Even
 	c.SetData(cloudevents.ApplicationJSON, commonEventData)
 	return c
 }
+
+// GetGracefulContext returns a context with cancel and a wait group to sync before shutdown
+func GetGracefulContext() context.Context {
```

Comment:
I'd rather make this unexported.
If you really want, you could still use it in the test as the package name is shared.